src/Main.elm

```diff
@@ -0,0 +1,139 @@
+module Main exposing (..)
+
+import Html exposing (..)
+import Html.App as App
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import Json.Decode as Json
+import String
+
+
+-- main
+
+main : Program Never
+main =
+  App.beginnerProgram
+    { model = emptyModel
+    , view = view
+    , update = update
+    }
+
+-- Model
+
+type alias Model =
+  { entries : List Entry
+  , input : String
+  , uid : Int
+  }
+
+type alias Entry =
+  { description : String
+  , id : Int
+  , taps : Int
+  }
+
+emptyModel : Model
+emptyModel =
+  { entries = []
+  , input = ""
+  , uid = 0
+  }
+
+
+-- Update
+
+type Msg
+  = NoOp
+  | UpdateInput String
+  | Add
+  | Tap Int
+
+
+update : Msg -> Model -> Model
+update msg model =
+  case msg of
+    NoOp -> model
+
+    UpdateInput str ->
+      { model | input = str }
+
+    Add ->
+      { model
+        | input = ""
+        , uid = model.uid + 1
+        , entries =
+            if String.isEmpty model.input then
+              model.entries
+            else
+              model.entries ++ [newEntry model.input model.uid]}
+
+    Tap id' ->
+      let updateEntry e =
+        if e.id == id' then { e | taps = e.taps + 1 } else e
+      in
+        { model | entries = List.map updateEntry model.entries }
+
+
+newEntry : String -> Int -> Entry
+newEntry desc id' =
+  { description = desc
+  , id = id'
+  , taps = 0
+  }
+
+onEnter : Msg -> Attribute Msg
+onEnter msg =
+  let
+    tagger code =
+      if code == 13 then msg else NoOp
+  in
+    on "keydown" (Json.map tagger keyCode)
```

Comment:
I get what's happening here, but what's up with the syntax? For starters, what is "on"? Is that a function from one of the imported modules? How is the keyCode actually "captured"?